libraries/chain/include/eosio/chain/name.hpp

```diff
@@ -179,9 +179,16 @@ namespace eosio::chain {
 namespace std {
    template<> struct hash<eosio::chain::name> : private hash<uint64_t> {
       typedef eosio::chain::name argument_type;
-      size_t operator()(const argument_type& name) const noexcept
-      {
-         return hash<uint64_t>::operator()(name.to_uint64_t());
+
+      size_t operator()(const argument_type& name) const noexcept {
+         static_assert(sizeof(size_t) == sizeof(uint64_t));
+         uint64_t v         = name.to_uint64_t();
+         auto     num_zeros = std::countr_zero(v);
+         if (num_zeros >= 4) {
+            uint64_t shift = 4 + ((num_zeros - 4) / 5) * 5;
+            return v >> shift;
+         }
+         return v;
```

Comment:
Seems overly complex to me -- a byteswap solves the same problem and resolves to just a single x86 instruction. Function can just be the static assert and `return bswap_64(name.to_int64_t())`

Reply:
Great, thanks! https://github.com/AntelopeIO/spring/pull/1206/commits/a4c82895f541f130ab51194578ccb6571a3e0549